datashader/transfer_functions/__init__.py

```diff
@@ -169,8 +170,10 @@ def eq_hist(data, mask=None, nbins=256*256):
 
     data2 = data if mask is None else data[~mask]
     if data2.dtype == bool or np.issubdtype(data2.dtype, np.integer):
-        if data2.dtype.kind == 'u':
-             data2 = data2.astype('i8')
+        # workaround for bincount not supporting uint64 (https://github.com/numpy/numpy/issues/17760)
```

Comment:
Out of curiosity, why not use `histogram` for `uint64` instead of downcasting it?

Reply:
It's a good question. Mainly I don't want to change too many things, as we're trying to get a release out by last week. But there are other factors, such as that if a user did use uint64, it would allocate an enormous array to hold the results, and I want to give users feedback not to do that. I also considered automatically switching to the binning approach (instead of one bin per possible integer value) for any 64-bit integer type, and on balance, maybe I should just do that.

Reply:
Philipp and I worked out a better approach using `np.unique` and putting guards on the potential number of unique values to avoid these issues.